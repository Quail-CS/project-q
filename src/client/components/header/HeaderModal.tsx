/**
 * @module HeaderModal.tsx
 * @description Application Header Link to Modal Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';

export interface HeaderModalProps {
  className: string,
  linkKey: string,
  children: string,
  onClick: () => any,
}

const HeaderModal = (props: HeaderModalProps): JSX.Element => {
  const { className, linkKey, onClick, children } = props;
  return (
    <li className={className} key={linkKey} onClick={onClick}>{children}</li>
  )
}

export default HeaderModal;
