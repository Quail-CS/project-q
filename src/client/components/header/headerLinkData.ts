/**
 * @module headerLinkData.ts
 * @description List of Header Links
 */

interface IHeaderLinkData {
  [linkName: string]: {
    className: string,
    linkKey: string,
    linkRoute?: string,
    linkText: string,
  }
}

export const headerLinkData: IHeaderLinkData = {
  logout: {
    className: 'header-right',
    linkKey: 'logout',
    linkRoute: '/account/logout',
    linkText: 'LOGOUT',
  },
  restart: {
    className: 'header-right',
    linkKey: 'restart',
    linkText: 'RESET',
  },
  screen: {
    className: 'header-right-screen',
    linkKey: 'screen',
    linkText: 'SCREEN YOUR PORTFOLIO',
  },
  home: {
    className: 'header-right',
    linkKey: 'home',
    linkRoute: '/',
    linkText: 'HOME',
  },
  register: {
    className: 'header-right',
    linkKey: 'register',
    linkRoute: '/account/register',
    linkText: 'REGISTER',
  },
  login: {
    className: 'header-right',
    linkKey: 'login',
    linkRoute: '/account/login',
    linkText: 'LOGIN',
  },
};
