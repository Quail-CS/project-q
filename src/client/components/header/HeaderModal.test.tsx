import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import HeaderModal from './HeaderModal';
import { headerLinkData } from './headerLinkData';

Enzyme.configure({ adapter: new Adapter() });

const onClick = jest.fn()

const props = {
  className: headerLinkData.login.className,
  linkKey: headerLinkData.login.linkKey,
  onClick: onClick,
}

describe('HeaderModal Component', () => {
  it('matches the snapshot', () => {
    const tree: Renderer.ReactTestRendererJSON = Renderer.create(
      <HeaderModal {...props}>{headerLinkData.login.linkText}</HeaderModal>
    ).toJSON() as Renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  })
  it('renders a <li> with key equal to linkKey prop', () => {
    const headerModalWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<HeaderModal {...props}>{headerLinkData.login.linkText}</HeaderModal>)
    expect(headerModalWrapper.type()).toEqual('li');
    expect(headerModalWrapper.key()).toEqual(props.linkKey);
    expect(headerModalWrapper.find('li').text()).toEqual(headerLinkData.login.linkText);
  })
  it('contains an onClick property which is called upon click', () => {
    const headerModalWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<HeaderModal {...props}>{headerLinkData.login.linkText}</HeaderModal>)
    headerModalWrapper.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  })
});
