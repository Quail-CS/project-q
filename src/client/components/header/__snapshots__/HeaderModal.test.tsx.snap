// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderModal Component matches the snapshot 1`] = `
<li
  className="header-right"
  onClick={[MockFunction]}
>
  LOGIN
</li>
`;
